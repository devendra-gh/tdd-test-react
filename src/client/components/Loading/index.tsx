import React from 'react';

function Loading() {
  return <div>Loading...</div>;
}

Loading.defaultProps = {};

export default Loading;
