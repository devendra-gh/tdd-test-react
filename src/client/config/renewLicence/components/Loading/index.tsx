import React from 'react';
import Spinner from '@tamm/ui-lib-v2-spinner';
import './index.less';

const Loading = () => (
  <div className="spinner-wrapper">
    <Spinner type="logo" />
  </div>
);

export default Loading;
