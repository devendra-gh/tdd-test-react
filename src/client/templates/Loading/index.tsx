// eslint-disable-next-line
import React from 'react';
import Spinner from '@tamm/ui-lib-v2-spinner';

export default () => {
  return (
    <div
      className="loadingOverlay" // Don't delete this class
      style={{
        position: 'fixed',
        zIndex: 5,
        top: 0,
        left: 0,
        width: '100%',
        height: '100%',
        display: 'flex',
        justifyContent: 'center',
        alignItems: 'center',
        overflow: 'hidden',
        background: 'rgba(255, 255, 255, 0.7)',
      }}
    >
      <Spinner type="logo" />
    </div>
  );
};
