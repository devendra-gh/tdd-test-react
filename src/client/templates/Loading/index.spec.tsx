import React from 'react';
import { shallow } from 'enzyme';
import Loading from './index';

describe('client/templates/Sidebar', () => {
  it('should render the loading template', () => {
    const component = shallow(<Loading />);
    expect(component).toMatchSnapshot();
  });
});
