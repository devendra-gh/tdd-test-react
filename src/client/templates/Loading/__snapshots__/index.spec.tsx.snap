// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`client/templates/Sidebar should render the loading template 1`] = `
<div
  className="loadingOverlay"
  style={
    Object {
      "alignItems": "center",
      "background": "rgba(255, 255, 255, 0.7)",
      "display": "flex",
      "height": "100%",
      "justifyContent": "center",
      "left": 0,
      "overflow": "hidden",
      "position": "fixed",
      "top": 0,
      "width": "100%",
      "zIndex": 5,
    }
  }
>
  <Spinner
    type="logo"
  />
</div>
`;
